plugins/services/src/js/structs/ServiceTree.js

```diff
@@ -201,10 +215,14 @@ module.exports = class ServiceTree extends Tree {
               return;
             }
 
-            let serviceLabels = service.getLabels();
-
+            let serviceLabels;
             if (service instanceof Service) {
-              serviceLabels = ServiceUtil.convertServiceLabelsToArray(service);
+              serviceLabels = Object.keys(service.getLabels()).map(key => ({
```

Comment:
you basically say those helper functions don't need to be tested anymore? (`convertServiceLabelsToArray `, `createServiceFromResponse `) . 
(i'm totally ok with that. just want to make sure.)

Reply:
well, no - BUT…
i tried refactoring it in a way that preserves the tests, but failed… I can see that some of the functions have been extracted for easier testing and thats a good idea in general - but extracting them into a util and not having that util pure is worse than not having them extracted IMHO… I tried to come up with a test testing this method, but failed in a reasonable timeframe…

not entirely sure if this change is "good", but I found it to be "good enough" 🤷‍♂️

Reply:
agree